modules/ipc/include/hephaestus/ipc/zenoh/internal/action_server_client_helper.h

```diff
@@ -0,0 +1,78 @@
+//================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <future>
+
+#include <fmt/core.h>
+
+#include "hephaestus/ipc/common.h"
+#include "hephaestus/ipc/subscriber.h"
+#include "hephaestus/ipc/zenoh/service.h"
+#include "hephaestus/ipc/zenoh/session.h"
+#include "hephaestus/ipc/zenoh/subscriber.h"
+#include "hephaestus/ipc/zenoh/types/action_server_types.h"
+#include "hephaestus/ipc/zenoh/types/action_server_types_protobuf.h"
+
+namespace heph::ipc::zenoh::internal {
+[[nodiscard]] auto getStatusPublisherTopic(const TopicConfig& server_topic) -> TopicConfig;
+
+[[nodiscard]] auto getResponseServiceTopic(const TopicConfig& topic_config) -> TopicConfig;
+
+[[nodiscard]] auto getStopServiceTopic(const TopicConfig& topic_config) -> TopicConfig;
+
+/// If an action server is already serving a request it will reject the new request.
```

Comment:
as mention above, I would add a way to also directly overwrite the running task (with request stop).

Reply:
yes I added a todo